tests/core/factory/test_checkpoint_factory.py

```diff
@@ -0,0 +1,238 @@
+from unittest.mock import Mock
+
+import pytest
+
+from great_expectations import set_context
+from great_expectations.checkpoint.checkpoint import Checkpoint
+from great_expectations.core.factory.checkpoint_factory import CheckpointFactory
+from great_expectations.data_context import AbstractDataContext
+from great_expectations.data_context.store.checkpoint_store import CheckpointStore
+from great_expectations.exceptions import DataContextError
+
+
+@pytest.fixture
+def checkpoint_dict():
+    return {
+        "name": "oss_test_checkpoint",
+        "expectation_suite_name": "oss_test_expectation_checkpoint",
+        "validations": [
+            {
+                "name": None,
+                "id": None,
+                "expectation_checkpoint_name": "taxi.demo_pass",
+                "expectation_checkpoint_ge_cloud_id": None,
+                "batch_request": None,
+            },
+        ],
+        "action_list": [
+            {
+                "action": {"class_name": "StoreValidationResultAction"},
+                "name": "store_validation_result",
+            },
+        ],
+    }
+
+
+def _assert_checkpoint_equality(actual: Checkpoint, expected: Checkpoint):
```

Comment:
Should this just be defined as `__eq__` on `Checkpoint?

Reply:
hmm maybe but I'm not sure how the structure of that object will change. I can follow up with a PR if we want